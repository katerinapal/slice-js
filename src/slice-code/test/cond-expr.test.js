import {runAllCombosTests, comboOfBools} from './helpers/utils'

// this is here to make it easy to isolate tests to a specific case
/*
test.only(
  'testing uncoveredConditionalExpression(false)',
  require('./helpers/utils').snapSlice(
    require.resolve('./fixtures/cond-expr'),
    ({uncoveredConditionalExpression}) => {
      uncoveredConditionalExpression(false)
    },
  ),
)
/* */

runAllCombosTests({
  filename: require.resolve('./fixtures/cond-expr'),
  methods: [
    {
      methodName: 'isConsequentOrAlternate',
      possibleArguments: comboOfBools(1),
    },
    {
      methodName: 'isConsequentOrAlternatesConsequentOrAlternate',
      possibleArguments: comboOfBools(2),
    },
    {
      methodName: 'uncoveredConditionalExpression',
      possibleArguments: comboOfBools(1),
    },
  ],
})
