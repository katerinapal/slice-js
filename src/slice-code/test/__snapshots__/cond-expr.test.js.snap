exports[`test isConsequentOrAlternate(false) && isConsequentOrAlternate(true) 1`] = `
"export { isConsequentOrAlternate };

function isConsequentOrAlternate(isConsequent) {
  return isConsequent ? isConsequent : !isConsequent;
}"
`;

exports[`test isConsequentOrAlternate(false) 1`] = `
"export { isConsequentOrAlternate };

function isConsequentOrAlternate(isConsequent) {
  return !isConsequent;
}"
`;

exports[`test isConsequentOrAlternate(true) && isConsequentOrAlternate(false) 1`] = `
"export { isConsequentOrAlternate };

function isConsequentOrAlternate(isConsequent) {
  return isConsequent ? isConsequent : !isConsequent;
}"
`;

exports[`test isConsequentOrAlternate(true) 1`] = `
"export { isConsequentOrAlternate };

function isConsequentOrAlternate(isConsequent) {
  return isConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(true, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, true) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(false, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(false, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(false, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent ? isAlternatesConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(false, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : !isConsequent && !isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) && isConsequentOrAlternatesConsequentOrAlternate(false, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent ? isConsequent : isAlternatesConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) && isConsequentOrAlternatesConsequentOrAlternate(true, false) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent;
}"
`;

exports[`test isConsequentOrAlternatesConsequentOrAlternate(true, true) 1`] = `
"export { isConsequentOrAlternatesConsequentOrAlternate };

function isConsequentOrAlternatesConsequentOrAlternate(isConsequent, isAlternatesConsequent) {
  return isConsequent;
}"
`;

exports[`test uncoveredConditionalExpression(false) && uncoveredConditionalExpression(true) 1`] = `
"export { uncoveredConditionalExpression };

function uncoveredConditionalExpression(cover) {
  if (cover) {
    return cover;
  }
  return !cover;
}"
`;

exports[`test uncoveredConditionalExpression(false) 1`] = `
"export { uncoveredConditionalExpression };

function uncoveredConditionalExpression(cover) {
  return !cover;
}"
`;

exports[`test uncoveredConditionalExpression(true) && uncoveredConditionalExpression(false) 1`] = `
"export { uncoveredConditionalExpression };

function uncoveredConditionalExpression(cover) {
  if (cover) {
    return cover;
  }
  return !cover;
}"
`;

exports[`test uncoveredConditionalExpression(true) 1`] = `
"export { uncoveredConditionalExpression };

function uncoveredConditionalExpression(cover) {
  return cover;
}"
`;
